paddle/fluid/operators/reduce_ops/mkldnn/reduce_mkldnn_op.h

```diff
@@ -121,5 +121,63 @@ class ReduceMKLDNNKernel : public framework::OpKernel<T> {
   }
 };
 
+template <typename T>
+class ReduceGradMKLDNNKernel : public framework::OpKernel<T> {
+ public:
+  void RunKernel(const framework::ExecutionContext& ctx,
+                 dnnl::algorithm binary_type, float scale_x,
+                 float scale_y) const {
+    auto& dev_ctx =
```

Comment:
Since you're not modifying dev_ctx:
```suggestion
    const auto& dev_ctx =
```